### ISSUE TITLE
Can't build using stack v1.4

### ISSUE THREAD

**Issue by dfordivam** [yi-editor/yi#975]
With this update to stack you will get the below error

"You cannot have packages and a shell-file filled at the same time in your nix-shell configuration."

Support reproducible builds via Nix (v1.4)
https://github.com/commercialhaskell/stack/commit/931cb02527a4d8dfe7652bdc44b144585a737b18

**Comment by ethercrow** [comment]
Works for me with stack-1.4.0 on macOS. Do you have to have nix installed to trigger the issue?

**Comment by dfordivam** [comment]
Yes it is specific to nix... and if you have nixos, you just cannot build yi...

**Comment by Fuuzetsu** [comment]
we don't have `packages` in top level stack.yaml, I suspect this ticket is out of date.

**Comment by dfordivam** [comment]
I still see packages in stack.yaml... and the issue with build is still there... 

flags: {}
nix:
  shell-file: shell.nix
packages:
- yi-core
- yi-dynamic-configuration
- yi-frontend-vty

**Comment by dfordivam** [comment]
How am I supposed to build yi ?
do a stack build at top level?

**Comment by dfordivam** [comment]
Some update..
I did `nix-shell` in top directory to enter the shell with required dependencies, then modified the yi/stack.yaml and added the local packages as in it. (like ../yi-core etc) 
Then I did stack build, and everything seems to go on fine, until I hit the issue #931 
I tried `LANG=C stack build`, but it did not help.

So I guess even if this issue is solved, I would still not be able to build yi.

**Comment by noughtmare** [comment]
Maybe you have `LC_ALL` set? That overrides the `LANG` option.

**Comment by Fuuzetsu** [comment]
somewhat related: yi was disabled in nixpkgs because it doesn't build there anymore; I don't plan to fix it in short term at least.
